lib/auth/auth_with_roles.go

```diff
@@ -1515,9 +1515,49 @@ func (a *ServerWithRoles) NewKeepAliver(ctx context.Context) (types.KeepAliver,
 	return nil, trace.NotImplemented(notImplementedMessage)
 }
 
-// GenerateUserCerts generates users certificates
-func (a *ServerWithRoles) GenerateUserCerts(ctx context.Context, req proto.UserCertsRequest) (*proto.Certs, error) {
-	return a.generateUserCerts(ctx, req)
+// GenerateInitialRenewableUserCerts generates renewable certs for a non-interactive user
+// using a previously issued single-use token.
+//
+// The token's TTL is enforced, and if the operation is successful the token is destroyed.
+func (a *ServerWithRoles) GenerateInitialRenewableUserCerts(ctx context.Context, req proto.RenewableCertsRequest) (*proto.Certs, error) {
+	if len(req.Token) == 0 {
+		return nil, trace.BadParameter("missing token")
+	}
+	if len(req.PublicKey) == 0 {
+		return nil, trace.BadParameter("missing public key")
+	}
+
+	token, err := a.authServer.GetUserToken(ctx, req.Token)
+	if err != nil {
+		log.Debugf("Could not fetch bot token: %+v", err)
+		return nil, trace.AccessDenied("the token is not valid")
+	}
+
+	if err := a.authServer.verifyUserToken(token, UserTokenTypeBot); err != nil {
+		return nil, trace.Wrap(err)
```

Comment:
Should this be `trace.AccessDenied`?

Reply:
`verifyUserToken` returns `trace.AccessDenied()` for the relevant denials, and I thought `trace.Wrap()` bubbled up the specific error? Almost all our other calls to `verifyUserToken` behave the same here, so I think it's okay?